### PR TITLE
add maven profile `deploy`, add rake suport

### PR DESCRIPTION
**Pull request by bs-jokri**
**1** maven profile `deploy` deploys all war files to the folder determined in
the property `deploy.dir`. One is able to deploy all files to the folder
`/ABSOLUTE/PATH/TO/deploy` via

```
mvn install -P deploy -DskipTests -Ddeploy.dir=/ABSOLUTE/PATH/TO/deploy
```

this is callable from all folders which contain targets to build war
files, i.e. in ./frontend

--

**2** add rake support for
- packaging via fpm to `.rpm`, `.deb` and `.tar.gz`
- compile sw360 within an docker container

### CODE REVIEW

**Comment by bs-jokri** [comment]
Please be aware of the following issue

This creates the folders
```
./frontend/sw360-theme/${env.LIFERAY_PATH}/
./frontend/layouttpl/${env.LIFERAY_PATH}/
./frontend/sw360-portlet/${env.LIFERAY_PATH}/
```
when LIFERAY_PATH is not set. This is a bad behaviour and should be repaired!
Solutions:
1. maven-liferay-plugin should not try to deploy on `mvn install` without `-Pdeploy`
2. recognize default case?

**Comment by maximilianhuber** [comment]
This pr removed the script `./scripts/install-thrift.sh` which is used by the travis test. This will be undone!

**Comment by maximilianhuber** [comment]
This pr needs to be updated by @bs-jokri.

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+
+ROOT=File.dirname(__FILE__)
+
+TARGET_PREFIX=ENV['TARGET_PREFIX'] || "/opt/sw360/deploy"
+TARGET_NAME="sw360"
+TARGET_VERSION="1.4.0-SNAPSHOT"
```
Comment (alex-evo) [inline Rakefile]: Is there a way to avoid hard-coding version?

File: sw360dev.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ && chmod +x /usr/local/bin/gosu \
+ && wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$(dpkg --print-architecture).asc" \
+ && export GNUPGHOME="$(mktemp -d)" \
+ && gpg --keyserver ha.pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 \
+ && gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu \
+ && rm -r "$GNUPGHOME" /usr/local/bin/gosu.asc && $_purge ca-certificates
+
+RUN set -x \
+ && echo "deb http://ftp.debian.org/debian jessie-backports main" > /etc/apt/sources.list.d/backports.list \
+ && $_update && $_install openjdk-8-jdk wget maven && $_cleanup \
```
Comment (alex-evo) [inline sw360dev.Dockerfile]: isn't `wget` installed by line 22?

File: Rakefile [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+DEV_CONTAINER_NAME = "sw360dev"
+DOCKERIZE=(ENV['DOCKERIZE'] || "true") == "true"
+
+print "########################################################################\n"
+print "# DEPLOY_DIR is set to: #{DEPLOY_DIR}\n"
+print "# MAVEN_PARAMETERS are set to: #{MAVEN_PARAMETERS}\n"
+print "# DOCKERIZE is set to: #{DOCKERIZE}\n"
+print "########################################################################\n\n"
+
+task :default => :install
```
Comment (alex-evo) [inline Rakefile]: I can't find where the task `:install` is defined. Rake agrees:
`rake aborted! 
Don't know how to build task 'install'`

Comment (maximilianhuber) [inline Rakefile]: You are right, install is removed since maven is capable of doing this.

Maybe `package` could be a new default

**Comment by maximilianhuber** [comment]
@bs-jokri: Please reset this PR to https://github.com/sw360/sw360portal/tree/dev/addDockerAndRakeSupport, which is rebased and fixes some of the mentioned issues

**Comment by maximilianhuber** [comment]
due to this merge there are changes necessary in the vagrant setup.

**Comment by alex-evo** [comment]
do you want to contribute that change, or would you rather want me to take care of it?

**Pull request merged**